File: 2962. Count Subarrays Where Max Element Appears at Least K Times/countSubarrays.cpp
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

class Solution {
public:
    long long countSubarrays(vector<int>& nums, int k) {
        long long res = 0;
        int maximum = *max_element(nums.begin(), nums.end());
        int l = 0;
        int sum = 0;


        for(int r=0; r<nums.size(); r++){
            nums[r] == maximum && sum++;

            while(sum >= k){
                res += (nums.size() - r);
                nums[l] == maximum && sum--;
                l++;
            }
        }
        return res;
    }
};

File: 2962. Count Subarrays Where Max Element Appears at Least K Times/countSubarrays.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var countSubarrays = function (nums, k) {
    let max = Math.max(...nums)

    let l = 0;
    let res = 0;
    let cnt = 0

    for (let r = 0; r < nums.length; r++) {
        if (nums[r] === max) {
            cnt++
        }

        while (cnt >= k) {
            res += (nums.length - r)
            if (nums[l] === max) cnt--
            l++
        }
    }

    return res;
};